boards/PDM/Test/Src/Test_StateMachine.cpp

```diff
@@ -61,25 +60,42 @@ class PdmStateMachineTest : public testing::Test
             send_non_periodic_msg_PDM_AIR_SHUTDOWN,
             send_non_periodic_msg_PDM_MOTOR_SHUTDOWN,
             send_non_periodic_msg_PDM_WATCHDOG_TIMEOUT);
-        can_rx_interface     = App_CanRx_Create();
-        vbat_voltage_monitor = App_VoltageMonitor_Create(
-            GetVbatVoltage, GetVbatMinVoltage, GetVbatMaxVoltage,
-            VbatErrorCallback);
-        _24v_aux_voltage_monitor = App_VoltageMonitor_Create(
-            Get24vAccVoltage, Get24vAccMinVoltage, Get24vAccMaxVoltage,
-            _24vAuxErrorCallback);
-        _24v_acc_voltage_monitor = App_VoltageMonitor_Create(
-            Get24vAccVoltage, Get24vAccMinVoltage, Get24vAccMaxVoltage,
-            _24vAccErrorCallback);
+        can_rx_interface   = App_CanRx_Create();
```

Comment:
Would be nice if we don't have to write all the `_Create` twice, once in `main` and once in `Setup`. What would be a nice way to do this @garethellis0 ?

Reply:
Sorry, not seeing where you're setting this up twice? `Setup` will run before each test, so that should be sufficient?

Reply:
Oh sorry when I said `main` I mean the main function in the ARM embedded target.
```
// main.c (ARM)
int main(void) {
  ..
  x = App_XXX_Create();
  y = App_YYY_Create();
  App_PdmWorld_Create(x, y);
  ..
}
```

```
// Test_StateMachine.cpp
void Setup override {
  ..
  x = App_XXX_Create();
  y = App_YYY_Create();
  App_PdmWorld_Create(x, y);
  ..
}
```
We're writing the same code for the embedded and  test targets

Reply:
Oh, I see. That's a bit of a tricky one, because we're not doing _exactly_ the same thing, in that we're passing in different IO functions in the tests versus in the arm target. I can't think of a neat solution here that wouldn't obfuscate things further.